packages/mobile-native-mcp-server/src/tools/magi/prd/magi-prd-gap-requirements/tool.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2025, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+import { McpServer } from '@modelcontextprotocol/sdk/server/mcp.js';
+import { Logger } from '../../../../logging/logger.js';
+import { GAP_REQUIREMENTS_TOOL, GapRequirementsInput } from './metadata.js';
+import { PRDAbstractWorkflowTool } from '../../../base/prdAbstractWorkflowTool.js';
+
+/**
+ * Tool for generating functional requirements based on identified gaps.
+ */
+export class MagiGapRequirementsTool extends PRDAbstractWorkflowTool<typeof GAP_REQUIREMENTS_TOOL> {
+  constructor(server: McpServer, logger?: Logger) {
+    super(server, GAP_REQUIREMENTS_TOOL, 'GapRequirementsTool', logger);
+  }
+
+  public handleRequest = async (input: GapRequirementsInput) => {
+    const guidance = this.generateGapRequirementsGuidance(input);
+    const finalOutput = this.finalizeWorkflowToolOutput(guidance, input.workflowStateData);
+    return finalOutput;
+  };
+
+  private generateGapRequirementsGuidance(input: GapRequirementsInput) {
+    const gapsList = input.identifiedGaps
+      .map(
+        (gap, index) => `${index + 1}. **${gap.title}** (${gap.severity} severity)
+- ID: ${gap.id}
+- Category: ${gap.category}
+- Description: ${gap.description}
+- Impact: ${gap.impact}
+- Suggested Requirements: ${gap.suggestedRequirements.map(req => req.title).join(', ')}`
+      )
+      .join('\n\n');
+
+    return `
+You are a product requirements analyst tasked with generating additional functional requirements for a Salesforce mobile native app based on identified gaps.
+
+## Feature Brief
+
+**File Path**: ${input.featureBriefPath}
+
+Please read the feature brief file from the path above.
+
+## Current Functional Requirements
+
+**File Path**: ${input.requirementsPath}
+
+Please read the requirements.md file from the path above.
+
+## Gap Analysis Context
+
+The following gaps have been identified in the current requirements that need to be addressed:
+
+${gapsList}
+
+## Your Task
+
+Your task is to generate NEW functional requirements that address the identified gaps. Focus on:
+
+1. **Addressing Critical Gaps**: Prioritize requirements that address critical and high-severity gaps
+2. **Building on Existing**: Ensure new requirements complement existing ones without duplication
+3. **Completeness**: Generate requirements that fill the identified gaps comprehensively
+4. **Integration**: Ensure new requirements integrate well with existing requirements
+
+**Important**: When analyzing existing requirements, focus on **approved requirements** and **modified requirements**. Ignore **rejected requirements** and **out-of-scope requirements** as they have been explicitly excluded from the feature scope.
```

Comment:
Nit: for readability in development, we probably need to establish a max column width for our string lines. I think my thumb in the air has traditionally been to line break around 90 chars, but we could just formally enforce it by configuring [`max-len`](https://eslint.style/rules/max-len) in our ESLint rules.

Reply:
agreed - I suggest we do that as a fast follow in case it touches a bunch across the project and creates a lot of noise